From d3b3d9b85c04c42286f56381687319379b09b069 Mon Sep 17 00:00:00 2001
From: Ramya Gnanasekar <quic_rgnanase@quicinc.com>
Date: Wed, 2 Nov 2022 10:27:17 +0530
Subject: [PATCH] mac80211: WAR to fix warning during AWGN simulation in
 repeater

6 GHz has different power level for different modes and
hence channel context will be different for AP and STA
which is expected.

During issue, since there are 2 channel context for AP and STA,
warning is thrown since AP thinks it as channel context for
different channels.

Fixing the issue through WAR: by commenting the warning.

Signed-off-by: Ramya Gnanasekar <quic_rgnanase@quicinc.com>

Index: b/net/mac80211/util.c
===================================================================
--- a/net/mac80211/util.c	2022-11-03 09:43:07.993093213 +0530
+++ b/net/mac80211/util.c	2022-11-03 09:46:40.575386832 +0530
@@ -4145,12 +4145,11 @@ void ieee80211_awgn_detected_work(struct
 	}
 	mutex_unlock(&local->chanctx_mtx);
 
-	if (num_chanctx > 1)
-		/* XXX: multi-channel is not supported yet */
-		WARN_ON_ONCE(1);
-	else
-		cfg80211_awgn_event(local->hw.wiphy, &chandef, GFP_KERNEL,
-				    local->chan_bw_interference_bitmap);
+	/* TODO: Validate number of channel context
+	 * since multi-channel is not supported.
+	 */
+	cfg80211_awgn_event(local->hw.wiphy, &chandef, GFP_KERNEL,
+			    local->chan_bw_interference_bitmap);
 }
 
 void ieee80211_dfs_radar_detected_work(struct work_struct *work)
